Cilsil/Cil/Parsers/RetParser.cs

```diff
@@ -29,13 +33,41 @@ internal class RetParser : InstructionParser
                         Expression returnVariable = new LvarExpression(
                             new LocalVariable(Identifier.ReturnIdentifier,
                                               state.Method));
-                        var retInstr = new Store(lvalue: returnVariable,
-                                                 rvalue: returnValue,
-                                                 type: Typ.FromTypeReference(retType),
-                                                 location: state.CurrentLocation);
-                        var retNode = new StatementNode(state.CurrentLocation,
-                                                        StatementNode.StatementNodeKind.ReturnStmt,
-                                                        state.ProcDesc);
+
+                        if (returnValue is BinopExpression)
```

Comment:
Have you tried a UnopExpression? Will this error happen in Unop case?

Reply:
Because we only observed this in the op_Inequality automatically-generated method with "ceq" binop expression, I wanted to make this bug fix specific to that (lest we make it more general than this and potentially silently create some kind of issue elsewhere).